test/lib/faktory/connection_test.exs

```diff
@@ -0,0 +1,41 @@
+defmodule Faktory.ConnectionTest do
+  use ExUnit.Case, async: false # Can't use async cuz Faktory.Tcp.Mock uses
```

Comment:
IMO, it would be worth passing a `name` param to `Mock` (or letting it run as an anonymous PID) so that we can run several in parallel.